python/CSXCAD/CSProperties.pyx

```diff
@@ -388,16 +389,30 @@ cdef class CSProperties:
         """
         return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRON, **kw)
 
-    def AddPolyhedronReader(self, filename, **kw):
-        """ AddPolyhedronReader(filename, **kw)
+    def AddPolyhedronReader(self, file:Path, **kw)->CSPrimitives:
+        """Add a polyhedron from file to this property.
 
-        Add a polyhedron from file to this property.
+        Arguments
+        ---------
+        file:
+            The path to the file to add.
+
+        Returns
+        -------
+        primitive:
+            An object of type `CSPrimitives` with the primitive just created.
 
         See Also
         --------
         CSXCAD.CSPrimitives.CSPrimPolyhedronReader : See here for details on primitive arguments
         """
-        return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRONREADER, filename=filename, **kw)
+        file = Path(file) # Make sure that whatever we receive can be interpreted as a path to a file.
+        if not file.is_file():
+            raise FileNotFoundError(f'`file` points to a file that does not exist. Received {repr(str(file))}, absolute path is {repr(str(file.resolve()))}. ')
+
+        file = file.resolve() # Convert the path into an absolute path, otherwise it fails silently later on.
```

Comment:
I'm not sure I like this. In some case I want to be relative (to the project) and not absolute? 
Again I do not like to take options from the user?

Reply:
The user can still provide a relative path to the argument `file`.